### PR TITLE
Generate graphs, when creating Doxygen documentation

### PR DESCRIPTION
**Pull request by havardAasen**
When building documentation with Doxygen, and `dot` is found, generate diagrams and graphs.

I changed the image format to svg, for smaller file size and better scaling.

`dot` is normally found in `graphviz` package and is entirely optional to install.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -469,6 +469,7 @@ message(STATUS " ")
 message(STATUS "Documentation:")
 message(STATUS "enabled: ${ENABLE_DOCS}")
 message(STATUS "doxygen: ${DOXYGEN_EXECUTABLE}")
+message(STATUS "graphviz: ${DOXYGEN_DOT_EXECUTABLE}")
```
Comment (evgenyz) [inline CMakeLists.txt:472]:
```suggestion
message(STATUS "dot (graphviz): ${DOXYGEN_DOT_EXECUTABLE}")
```

**Comment by evgenyz** [comment]
@jan-cerny What do you thing about diagrams in our API docs?

File: CMakeLists.txt (hunk 1 above)
Comment (havardAasen) [reply CMakeLists.txt]: Updated with the applied suggestion

**Comment by jan-cerny** [comment]
@evgenyz They look like they won't help you much, you would need some more knowledgable diagram, that would have to be created manually I think.

**Review by jan-cerny — APPROVED** [review]
Thank you!

**Pull request merged**